sql/rename.go

```diff
@@ -209,6 +213,8 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
 	if err := p.txn.Put(descKey, tableDesc); err != nil {
 		return nil, err
 	}
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
the filter in sql/main_test does a pretty good job of checking all intents on the EndTransactionRequest. Combined with the logic tests (which I hooked into), we get nice coverage. eg: see the `testdata/txn` change that exercises the trigger persistence.
As for the actual trigger, I added this in `server/server_test` for lack of a better place (I really wanted to use the test server). More will tests will surely come as I make use of the actual gossipped data.